File: Dateien/de/__NeuGutesDeutsch/Umsetzer.js
function sz(was) { return was.replace(/ss/g, 'ß'); } 
function uc(was) { return was.toUpperCase(); }  
function lc(was) { return was.toLowerCase(); }  

function filtereRechtschreibung(was)  {
  was = was.replace(/\bAss\b/g, 'As');
  was = was.replace(/\bAlbtraum\b/g, 'Alptraum');
  was = was.replace(/sodass\b/g, 'so dass');
  was = was.replace(/(S|s)tattdessen/g, '$1tatt dessen');
  was = was.replace(/(A|a)ufw.ndig/g, '$1ufwendig');
  was = was.replace(/elbstständig/g, 'elbständig');
  was = was.replace(/([Nn])ummerier/g, '$1umerier');
  was = was.replace(/([Pp])latzier/g, '$1lazier');
  was = was.replace(/([Pp])otenziell/g, '$1otentiell');
  was = was.replace(/([Pp])otenzial/g, '$1otential');
  was = was.replace(/(E|e)in jedes Mal/g, '$1in jedes# Mal');   //  schützen
  was = was.replace(/(B|b)isschen/g, '$1ißchen');
  was = was.replace(/(J|j)edes Mal/g, '$1edesmal');
  was = was.replace(/(V|v)oran treiben/g, '$1orantreiben');     
  was = was.replace(/ssch/g, 's#sch');                 //  Seltenes schützen
  was = was.replace(/Ausstieg/g, 'Aus#stieg');         //  Seltenes schützen
  was = was.replace(/Bussteig/g, 'Bus#steig');         //  Seltenes schützen
  was = was.replace(/undesstaat/g, 'undes#staat');     //  Seltenes schützen
  was = was.replace(/(au)ss([aeä])/gi, '$1s#s$2');     //  Seltenes schützen
  was = was.replace(/sszene/gi, 's#szene');            //  Seltenes schützen
  was = was.replace(/sstunde/gi, 's#stunde');          //  Seltenes schützen
  was = was.replace(/(s|S)trasse/g, '$1traße');
  was = was.replace(/(h|H)asserf/g, '$1aßerf');
  nix = was.replace(/(Fassentnahme)/gm,   sz(RegExp.$1));
  was = was.replace(/(Fassentnahme)/gm,   sz(RegExp.$1));
  NIX = was.replace(/(Messergebnis)/gm,   sz(RegExp.$1));
  was = was.replace(/(Messergebnis)/gm,   sz(RegExp.$1));
  NIX = was.replace(/(Passersatz)/gm,     sz(RegExp.$1));
  was = was.replace(/(Passersatz)/gm,     sz(RegExp.$1));
  NIX = was.replace(/(schlussendlich)/gm, sz(RegExp.$1));
  was = was.replace(/(schlussendlich)/gm, sz(RegExp.$1));
  was = was.replace(/reformierter/g, 'herkömmlicher = moderner');   //  Diese Zeile nur für diese Zeige-Fassung!
  was = was.replace(/ss\b/g, 'ß');                      //  ss am Wortende  ->  ß
  was = was.replace(/sss/g, 'ßs');                      //  sss -> ßs
  was = was.replace(/ussa/ig, 'ußa');                    //  essta  -> eßta
  was = was.replace(/([mM])issacht/g, '$1ißacht');                      //  ss am Wortende  ->  ß
  NIX = was.replace(/((?:[^GB]|\A)[bcdfghjklmnpqrstvwxyz][aeiouäöü])ss([bcdfghjklmnpqrstvwxyz])/ig, '$1ß$2');  // ss >  ß ;  Glasstück, Grasstück
  was = was.replace(/((?:[^GB]|\A)[bcdfghjklmnpqrstvwxyz][aeiouäöü])ss([bcdfghjklmnpqrstvwxyz])/ig, '$1ß$2');  // ss >  ß ;  Glasstück, Grasstück
  NIX = was.replace(/([eio])\1\1/g, '$1$1-' + uc(RegExp.$1));            //  Teeei ->Tee-Ei
  was = was.replace(/([eio])\1\1/g, '$1$1-' + uc(RegExp.$1));            //  Teeei ->Tee-Ei
  was = was.replace(/(.)\1\1([aeiouyäöü])/g, '$1$1$2'); //  mmme -> mme
  NIX = was.replace(/(\d+)-(jähr|mal|fach|er)/g, lc(RegExp.$1+RegExp.$2));
  was = was.replace(/(\d+)-(jähr|mal|fach|er)/g, lc(RegExp.$1+RegExp.$2));
  was = was.replace(/so genannt/g, 'sogenannt')
  was = was.replace(/(S|s)elbstständig/, '$1elbständig');
  was = was.replace(/rgendetwas/g, 'rgend etwas');
  was = was.replace(/rgendeiner/g, 'rgend einer');
  was = was.replace(/(z)urzeit/g, '$1ur Zeit');  //zurzeit, Zurzeit
  was = was.replace(/(G|g)räulich/g, '$1reulich');
  was = was.replace(/Tipp\b/g, 'Tip');
  NIX = was.replace(/(lles beim Alten)/g,            lc(RegExp.$1));
  was = was.replace(/(lles beim Alten)/g,            lc(RegExp.$1));
  NIX = was.replace(/([Ii])(m Allgemeinen)/g,  '$1' +lc(RegExp.$2));
  was = was.replace(/([Ii])(m Allgemeinen)/g,  '$1' +lc(RegExp.$2));
  NIX = was.replace(/([Ii])(m Wesentlichen)/g, '$1' +lc(RegExp.$2));
  was = was.replace(/([Ii])(m Wesentlichen)/g, '$1' +lc(RegExp.$2));
  was = was.replace(/#/g, '');    //  # ist zeitweiliger ss-Schützer
  return (was);
  }

function erneuereRechtschreibung()  {
  nix = filtereRechtschreibung(document.getElementById('Reformschrieb').value);  //    zum Aufwärmen, damit RegExp sich selbst kennt, seufz ....
  document.getElementById('Rechtschrieb').value = filtereRechtschreibung(document.getElementById('Reformschrieb').value);
  }

File: Dateien/de/__NeuGutesDeutsch/Umsetzer/Umsetzer.pl
#!/usr/bin/perl
#########################################################################################################
##    Der Umsetzer  Doofschreib -> Rechtschreib
#########################################################################################################
$/ = chr(10);
#$\ = "\n<br>";

use CGI qw/:standard/;
print "content-type: text/html\n\n";
print '<head></head><body>';

$q = new CGI;
foreach $c($q->param)  {
     $q{$c} = $q->param($c)  }

$_ = $q{Reformschrieb};

### ==
sub sz { $Kt = shift; $Kt =~ s/ss/ß/;  $Kt }  ##  Verfahren: Seltenes umsetzen

s,\bAss\b,As,g;
s,\bAlbtraum\b,Alptraum,g;
s,sodass\b,so dass,g;
s,(S|s)tattdessen,$1tatt dessen,g;
s,(A|a)ufw.ndig,$1ufwendig,g;
s,([Nn])ummerier,$1umerier,g;
s,([Pp])latzier,$1lazier,g;
s,([Pp])otenziell,$1otentiell,g;
s,([Pp])otenzial,$1otential,g;
s((E|e)in jedes Mal)($1in jedes# Mal)g;   ##  schützen
s((B|b)isschen)($1ißchen)g;
s((J|j)edes Mal)($1edesmal)g;
s((V|v)oran treiben)($1orantreiben)g;     
s(ssch)(s#sch)g;                 ##  Seltenes schützen
s(Ausstieg)(Aus#stieg)g;         ##  Seltenes schützen
s(Bussteig)(Bus#steig)g;         ##  Seltenes schützen
s(Bundesstaat)(Bundes#staat)g;   ##  Seltenes schützen
s((s|S)trasse)($1traße)g;
s((h|H)asserf)($1aßerf)g;
s,(Fassentnahme|Messergebnis|Passersatz|schlussendlich),sz($1),ge;    ## Seltenes umsetzen
s/reformierter/herkömmlicher = moderner/;   ##  Diese Zeile nur für diese Zeige-Fassung!
s/ss\b/ß/g;                      ##  ss am Wortende  ->  ß
s/sss/ßs/g;                      ##  sss -> ßs
s/ussa/ußa/g;                    ##  essta  -> eßta
s/([mM])issacht/$1ißacht/g;                      ##  ss am Wortende  ->  ß
s/((?:[^GB]|\A)[bcdfghjklmnpqrstvwxyz][aeiouäöü])ss([bcdfghjklmnpqrstvwxyz])/$1ß$2/gi;  ##  Glasstück, Grasstück
s,([eio])\1\1,"$1$1-".uc($1),ge;            ##  Teeei ->Tee-Ei
s/(.)\1\1([aeiouyäöü])/$1$1$2/g; ##  mmme -> mme
s/(\d+)-(jähr|mal|fach|er)/lc("$1$2")/gei;
s/so genannt/sogenannt/g;
s,(S|s)elbstständig,$1elbständig,g;
s,rgendetwas,rgend etwas,g;
s,rgendeiner,rgend einer,g;
s/zurzeit/zur Zeit/g;
s/(G|g)räulich/$1reulich/g;
s,Tipp\b,Tip,g;
s/(lles beim Alten)/lc($1)/ge;
s/([Ii])(m Allgemeinen)/$1.lc($2)/ge;
s/([Ii])(m Wesentlichen)/$1.lc($2)/ge;
s/#//g;

$Vorlage = "../html/Umsetzer/Umsetzer.html";
open D_E, "$Vorlage" or print "nicht gefunden: $Vorlage ";
$Inhalt = join "", <D_E>;

$Inhalt =~ s,Hier bitte Reformschrieb eintragen .*?Schluss\.,$q{Reformschrieb},s;
$Inhalt =~ s,Hierher kommt dann das Ergebnis\.,$_,;

print $Inhalt;

print "\n<br>
<br>- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - <br>
Die Werte:
<br>- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - <br>";
for (sort keys %q)  {
  print "\n<br>$_ <b>$q{$_}</b>"  }

